src/api/app/models/user.rb

```diff
@@ -178,12 +178,17 @@ def self.update_notifications(params, user = nil)
     end
   end
 
-  def self.create_user_with_fake_pw!(attributes = {})
+  def self.create_user_with_fake_pw(attributes = {})
     chars = ["A".."Z", "a".."z", "0".."9"].collect(&:to_a).join
     fakepw = (1..24).collect { chars[rand(chars.size)] }.pack("a" * 24)
 
     attributes[:password] = fakepw
-    create!(attributes)
+    create(attributes)
+  end
+
+  def self.create_user_with_fake_pw!(attributes = {})
+    user = create_user_with_fake_pw(attributes)
+    user.valid? ? user : user.save!
```

Comment:
ActiveRecord [create](https://apidock.com/rails/ActiveRecord/Base/create/class) already saves the object to the database so why does it need to be saved again here?